### PR TITLE
Add Debug Output Copy All command : Fixes #27079

### LINKED ISSUE #27079: Add Command: Keyboard assignable DebugOutput Copy All
Add complementary keyboard command to copy all debug output.
Use new base from recently added Context Menu option

### PR DESCRIPTION
**Pull request by cleidigh**
@isidorn 
Global CopyAll Debug output

### CODE REVIEW

File: src/vs/workbench/parts/debug/electron-browser/repl.ts [hunk 1]
```diff
@@ -375,3 +390,26 @@ CommonEditorRegistry.registerEditorCommand(new SuggestCommand({
...
 	}
 }));
+
+@editorAction
+class ReplCopyAllAction extends EditorAction {
+
+	constructor() {
+		super({
+			id: 'repl.action.copyall',
+			label: nls.localize('actions.repl.copyall', "Debug Copy All"),
```
Comment (isidorn) [inline src/vs/workbench/parts/debug/electron-browser/repl.ts]: I would prefer the title to be "Debug: Console Copy All", that way it nicely aligns with the other debug actions and has a more clear name

File: src/vs/workbench/parts/debug/electron-browser/repl.ts [hunk 2]
```diff
@@ -375,3 +390,26 @@ CommonEditorRegistry.registerEditorCommand(new SuggestCommand({
...
+			alias: 'Debug Copy All',
+			precondition: debug.CONTEXT_IN_DEBUG_REPL,
+			kbOpts: {
```
Comment (isidorn) [inline src/vs/workbench/parts/debug/electron-browser/repl.ts]: I think you can just leave out the `kbOpts` and it should nicely compile, please try that

File: src/vs/workbench/parts/debug/electron-browser/repl.ts [hunk 3]
```diff
@@ -230,6 +232,19 @@ export class Repl extends Panel implements IPrivateReplService {
 		this.layout(this.dimension);
 	}
```
Comment (isidorn) [inline src/vs/workbench/parts/debug/electron-browser/repl.ts]: I do not like code duplication, can you please look into removing the `CopyAllAction` from `electronDebugActions.ts`. I believe that we can now always use this action instead, can you please try that?

**Comment by isidorn** [comment]
@cleidigh great work. I have added some comments directly in the commit which I would like you to address. Apart from that did you test this out, does it work nicely?

**Comment by cleidigh** [comment]
@isidorn 
made the two tweaks, compiles and executes fine.
I was thinking the same thing about using the same function
I'll work on that.

**Comment by cleidigh** [comment]
@isidorn 
didn't make much progress yet.
I somehow have to coerce the action registration for the ContextMenu IAction
actions.push(new CopyAllAction(CopyAllAction.ID, CopyAllAction.LABEL, tree));
Should be aligned with the ieditor action call
class ReplCopyAllAction extends EditorAction {
the signatures and call methods are different enough that I have not figured out
the right way to do it other than the action: calling the other one which I don't think is the right answer.

**Comment by isidorn** [comment]
@cleidigh great for the two tweaks, feel free to push those.
As for the action duplication, I do not understand why in the `replViewer` `getSecondayActions` you can not simply add a new instance of your new `EditorAction` since `EditorActions` are also `IActions`.

Though after pushing those two twearks we can merge it in and I can look into the last step of combining the two actions.

And yeah calling the other via `commandService` one we can leave as plan b

**Comment by cleidigh** [comment]
@isidorn 
I tried doing the instantiation. I can't get an acceptable signature.
I think the editor action with the @EditorAction decorator might be the problem.
pushed the other two tweaks. I'll continue to work on the duplication removal.

**Comment by cleidigh** [comment]
@isidorn 
well I'm stymied. 
I have tried several different approaches, 
- using the service successor in copy all action - type visibility problem - could not get to compile
- placing the copy all action in repl.ts and it calling the service successor - create some sort of Loop
- using the command service to call the editor action - compiles and runs, the function is called
but does not work


export class CopyAllAction extends Action {
	static ID = 'workbench.debug.action.copyAll';
	static LABEL = nls.localize('copyAll', "Copy All");
	private _commandService: ICommandService;

	constructor(id: string, label: string,
		@ICommandService commandService: ICommandService
		) {
		super(id, label);
		this._commandService = commandService;
	}

	public run(): TPromise<any> {
		this._commandService.executeCommand('repl.action.copyall');
		console.log('CopyAll eda');
		return TPromise.as(null);
	}
}

bugs me I can't figure it out !!

**Comment by isidorn** [comment]
@cleidigh thanks a lot for trying it out, great work! Let's merge this in and then I will try to remove code duplication

**Comment by isidorn** [comment]
@cleidigh this turned out to be more complicated than expected.
I have pushed some minor polish on top of your PR, however I was also running into the same issues as you did while you tried to reduce code duplicaton.
In your case the command does not get exectued from the context menu because it's precondition is not met.

**Comment by cleidigh** [comment]
@isidorn 
looks good
so you're just leaving the duplicate code for now?

**Comment by isidorn** [comment]
@cleidigh will fix it once I find time, currently working on multi-root features.
Thanks for understandgin

**Pull request merged**